### DISCUSSION TITLE
How to control the display and hide of the Badge component

### DISCUSSION THREAD

**Discussion by dxvgef** [primefaces/community#2651] [PrimeVue]
In PrimeVue v4, the `OverlayBadge` component replaces the `v-badge` directive, for example:
```vue
<OverlayBadge>
    <div id="test">...</div>
</OverlayBadge>
```
However, in some scenarios, I don't want this `<div>` to have any badge effect. When I add `v-if` or `v-show` to the `<OverlayBadge>` to control its visibility, it also hides the `<div>` inside. I only want to remove the badge effect from the `<div>`. Is there any other solution?

**Comment by dxvgef** [comment]
Please help me

**Comment by FeBe95** [comment]
I guess you could create a new `*.vue` file which contains the div. Then, in your main component, include it twice (with `v-if` directives).

**Comment by Eronne** [comment]
Would be really great if this `BadgeOverlay` handle the v-if internally.
